src/Features/Core/Portable/Diagnostics/DiagnosticArguments.cs

```diff
@@ -20,52 +22,61 @@ internal class DiagnosticArguments
         /// Additionally, diagnostic computation for explicit user commands, such as FixAll,
         /// are marked high priority.
         /// </summary>
+        [DataMember(Order = 0)]
         public bool IsHighPriority;
 
         /// <summary>
         /// Flag indicating if suppressed diagnostics should be returned.
         /// </summary>
+        [DataMember(Order = 1)]
         public bool ReportSuppressedDiagnostics;
 
         /// <summary>
         /// Flag indicating if analyzer performance info, such as analyzer execution times,
         /// should be logged as performance telemetry.
         /// </summary>
+        [DataMember(Order = 2)]
         public bool LogPerformanceInfo;
 
         /// <summary>
         /// Flag indicating if the analyzer telemety info, such as registered analyzer action counts
         /// and analyzer execution times, should be included in the computed result.
         /// </summary>
+        [DataMember(Order = 3)]
         public bool GetTelemetryInfo;
 
         /// <summary>
         /// Optional document ID, if computing diagnostics for a specific document.
         /// For example, diagnostic computation for open file analysis.
         /// </summary>
+        [DataMember(Order = 4)]
         public DocumentId? DocumentId;
 
         /// <summary>
         /// Optional document text span, if computing diagnostics for a specific span for a document.
         /// For example, diagnostic computation for light bulb invocation for a specific line in active document.
         /// </summary>
+        [DataMember(Order = 5)]
         public TextSpan? DocumentSpan;
 
         /// <summary>
         /// Optional <see cref="AnalysisKind"/>, if computing specific kind of diagnostics for a document request,
         /// i.e. <see cref="DocumentId"/> must be non-null for a non-null analysis kind.
         /// Only supported non-null values are <see cref="AnalysisKind.Syntax"/> and <see cref="AnalysisKind.Semantic"/>.
         /// </summary>
+        [DataMember(Order = 6)]
         public AnalysisKind? DocumentAnalysisKind;
 
         /// <summary>
         /// Project ID for the document or project for which diagnostics need to be computed.
         /// </summary>
+        [DataMember(Order = 7)]
         public ProjectId ProjectId;
 
         /// <summary>
         /// Array of analyzer IDs for analyzers that need to be executed for computing diagnostics.
         /// </summary>
+        [DataMember(Order = 8)]
```

Comment:
aside, would be nice to have an analzyer to prevent mistakes here :)

Reply:
Yes, it would be nice. Since Data Contracts are in BCL I believe this would be up to the BCL team. @mavasani are you aware of any FxCop/BCL analyzers validating data contract attributes?

Reply:
I don't think so, I'll file a feature request on runtime repo.

Reply:
@mavasani Thanks!

Reply:
https://github.com/dotnet/runtime/issues/42412